test/e2e/framework/resource/resources.go

```diff
@@ -62,10 +63,17 @@ func ScaleResource(
 }
 
 // DeleteResourceAndWaitForGC deletes only given resource and waits for GC to delete the pods.
+// This is a wrapper around DeleteResourceAndWaitForGCWithCustomResourceClient.
 func DeleteResourceAndWaitForGC(c clientset.Interface, kind schema.GroupKind, ns, name string) error {
+	return DeleteResourceAndWaitForGCWithCustomResourceClient(c, nil, nil, kind, ns, name)
+}
+
+// DeleteResourceAndWaitForGCWithCustomResourceClient deletes only given resource and waits for GC to delete the pods.
+// Enables to provide a custom resourece client, e.g. to fetch a CRD object.
+func DeleteResourceAndWaitForGCWithCustomResourceClient(c clientset.Interface, scaleClient scaleclient.ScalesGetter, resourceClient dynamic.ResourceInterface, kind schema.GroupKind, ns, name string) error {
```

Comment:
passing a resourceClient to this function that is only valid to delete a single custom resource doesn't make a lot of sense... you can pass a dynamic.Interface that can delete any resource